app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt

```diff
@@ -40,20 +43,59 @@ class DistanceExpressionActivity : AppCompatActivity() {
         +geoJsonSource(CIRCLE_ID) {
           geometry(circle)
         }
+        +geoJsonSource(POINT_ID2) {
+          geometry(center2)
+        }
+        +geoJsonSource(CIRCLE_ID2) {
+          geometry(circle2)
+        }
         +layerAtPosition(
           fillLayer(CIRCLE_ID, CIRCLE_ID) {
             fillOpacity(0.5)
             fillColor("#3bb2d0")
           },
           below = POI_LABEL
         )
+        +layerAtPosition(
+          fillLayer(CIRCLE_ID2, CIRCLE_ID2) {
+            fillOpacity(0.5)
+            fillColor("#00b2d0")
+          },
+          below = POI_LABEL
+        )
       }
     ) {
-      val symbolLayer = it.getLayer("poi-label") as SymbolLayer
+      val symbolLayer = it.getLayer(POI_LABEL) as SymbolLayer
+
+      var startNano = System.nanoTime()
+      val lineString = LineString.fromJson(LINESTRING_JSON)
+      println("Line string init : ${System.nanoTime() - startNano}")
+
+      val iter = 100
+      startNano = System.nanoTime()
+      for (i in 0..iter) {
+        Expression.distanceFast(lineString)
+      }
+      println("Distance fast time : ${System.nanoTime() - startNano}")
+
+      startNano = System.nanoTime()
+      for (i in 0..iter) {
+        Expression.distance(lineString)
+      }
+      println("Distance time : ${System.nanoTime() - startNano}")
+
+      startNano = System.nanoTime()
+      for (i in 0..iter) {
+        Expression.fromRaw(LINESTRING_JSON)
+      }
+      println("From raw : ${System.nanoTime() - startNano}")
```

Comment:
Nit: ICYMI you can use Kotlin [measureNanoTime](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.system/measure-nano-time.html)
```suggestion
val timeInNanos = measureNanoTime {
      repeat(iter) {
        Expression.fromRaw(LINESTRING_JSON)
      }
}
println("From raw : ${timeInNanos}")
```